### ISSUE TITLE
Color/message doesn't appear in all screens

### ISSUE THREAD

**Issue by Barbarossa93** [bakkeby/slock-flexipatch#4]
As discussed in #3, the message from the colormessage or message patch will always show on the INPUT screen, but doesnt always show on the INIT screen, and never shows on the PAM screen. Unfortunately, only my system seems to be affected by this so far as @bakkeby is unable to reproduce.

**Comment by Barbarossa93** [comment]
I decided to try a clean slock from suckless repo's directly. Previously when I said I did a clean build I did a clean build of slock-flexipatch. I have the exact same problem. @bakkeby can you tell me what version of libxinerama you use? I assume that is the library needed for the patch

**Comment by bakkeby** [comment]
I have
```
libxinerama 1.1.4-3
```

**Comment by bakkeby** [comment]
Do you have the same problem if you take a bare slock build and apply the message patch on top of that?

**Comment by Barbarossa93** [comment]
Bummer, I have the same version of libxinerama.

Yes, I still have the same problem with a bare slock with just the message patch

**Comment by Barbarossa93** [comment]
Do you know what part of the patch decides when to draw the text? I'm looking at it and I'm unsure if the variable is `screen`, `win`,or something else. Further, do you run a multimonitor setup? I've noticed that the message doesn't appear on the external monitor is plugged in and it seems to show up more often on the main monitor

**Comment by Barbarossa93** [comment]
Adding the `0` worked!! The message still disappears during PAM verification, but that's not a big deal to me. I haven't tested on an external monitor yet, so we may want to wait a bit before we close this

**Comment by Barbarossa93** [comment]
Well this is interesting, running `slock` within a window manager it works. Outside of a WM, it doesn't. This is important to me since I use `.xinitrc` to run the lock first while things get setup in the background. I'll try to do some investigating

**Comment by bakkeby** [comment]
I guess using a login manager would also make a difference (I use one).

**Comment by Barbarossa93** [comment]
So I altered the main function in slock.c where the `writemessage` function is called so that the `if` line just before it should always execute. Made no difference, but with changing my program startup order, I've found that if picom starts after slock, the message disappears. This is presumably because picom is reparenting the screen and therefore wiping the message. Additionally, my WM, herbstluftwm, has an option to "lock" the monitor. When the monitor is locked, reparenting is prevented. It seems like the logical thing to do is to start slock while the WM is locked and then start picom. Unfortunately, while the WM is locked, the message doesn't appear. I think what might be the best solution for my situation is to somehow redraw the message whenever the screen is reparented, although that seems hacky and probably overly complicated

**Comment by Barbarossa93** [comment]
*repainting, not reparenting

**Comment by bakkeby** [comment]
What do you use to trigger slock? I use `xautolock`.

**Comment by Barbarossa93** [comment]
On startup in my `.xinitrc`, I call it directly with a wrapper script. Otherwise, I use `xss-lock` to lock my session normally. Note that if I call the wrapper script directly within the session I don't have any issues. The only time I don't see the message immediately is during startup

**Comment by bakkeby** [comment]
Sounds like there is a race condition there. How / when do you start picom? Could you not start slock shortly after starting picom? Not sure if it works if you spawn a new process (script) that has a sleep before slock is run.

As I understand it you are using slock as a kind of bare boned login manager.

**Comment by Barbarossa93** [comment]
Yes, sorry if I should've been more clear about that earlier. I'm using slock as a "login manager" by having it be one of the first things that pop up during boot. So I think I finally figured out the optimal conditions: I have to place slock within my WM autostart because It wont show the message at all if it starts before the WM. Further, I had to introduce a one second delay just before the lock after doing a ton of trial and error to see which of my startup programs was causing the issue. Turns out my WM is doing something that causes the message to get wiped even after the WM "unlocks". Its perhaps not the most ideal solution, but adding 
```
trap '' INT TSTP
```
to both my `.xinitrc` and WM's autostart make me feel a little more secure about having my locker show up late. To answer your question, I had started picom very early on the startup process and then the lock much later, so I don't think picom was the only thing affecting the lock.  Unless we can make slock write it's message permanently (i.e. not susceptible to repainting events) I think this is effectively solved. It's some weird quirk of my setup that seems to be causing this issue.
